### DISCUSSION TITLE
using query raw with like '%%' got error N/A

### DISCUSSION THREAD

**Discussion by danieltan007** [prisma/prisma#20568] [Q&A] [answered]
### Question

hello
im testing out my prisma query raw like this:
```js
const data = await prisma.$queryRaw`
		SELECT
		  tbbarang.barang_id,
		  tbbarang.nama_barang,
		  tbbarang.golongan,
		  tbkategoriproduk.namakategori,
		  tbbarang.harga_beli,
		  tbvendor.nama,
		  tbvendor.alamat
		FROM tbbarang
		  INNER JOIN tbkategoriproduk ON tbbarang.jenis = tbkategoriproduk.idkategori
		  INNER JOIN tbvendor ON tbbarang.supplier = tbvendor.kodevendor
			where tbbarang.nama_barang like '%${namaBarang}%'
		ORDER BY 
		  tbbarang.nama_barang asc`;

		return data;

```
and i got error: code: 'N/A', message: 'N/A'

its only works if not using where like clause

im using MySQL database with prisma 5.1.1

thanks in advance

**Comment by ludralph** [comment]
Hi @danieltan007 👋 

Thank you for raising this question.

it looks like the issue is with the usage of the `namaBarang` parameter in the query. To resolve this, you should use parameterized queries. For example:
```js
const data = await prisma.$queryRaw`
    SELECT
      tbbarang.barang_id,
      tbbarang.nama_barang,
      tbbarang.golongan,
      tbkategoriproduk.namakategori,
      tbbarang.harga_beli,
      tbvendor.nama,
      tbvendor.alamat
    FROM tbbarang
      INNER JOIN tbkategoriproduk ON tbbarang.jenis = tbkategoriproduk.idkategori
      INNER JOIN tbvendor ON tbbarang.supplier = tbvendor.kodevendor
    WHERE tbbarang.nama_barang LIKE '%' || ${namaBarang} || '%'
    ORDER BY tbbarang.nama_barang ASC
  `;

  return data;
```
In the updated code, we use `|| ${namaBarang} ||` to properly concatenate the parameter value in the query. This way, we ensure that the `namaBarang` parameter is safely used in the query without introducing SQL injection vulnerabilities. Please try this updated code, and it should fix the issue you were facing.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by danieltan007** [reply, marked as answer]
currently im using  `CONCAT('%', ${namaBarang}, '%')`, after tried based on your code, its not filtering, it just showing all the data

**Reply by dsolis-ef** [reply]
thank you very useful

**Reply by dominiwe** [reply]
I just want to mention a small thing here for anyone finding this. If `namaBarang` is user supplied, note that the user can use `%` and `_` in the query. Even if not a security issue, the fact that these wildcards work could be unexpected behaviour for the user.
